public/typescript/domFunctions.ts

```diff
@@ -0,0 +1,48 @@
+import { curryN } from 'ramda';
+
+/**
+ *
+ * @example
+ * <table>
+ *   <tr>
+ *     <th>Name</th>
+ *     <td>Foo</td>
+ *   </tr>
+ *   <tr>
+ *     <th>Last name</th>
+ *     <td>Bar</td>
+ *   </tr>
+ * </table
+ *
+ * Return set with:
+ *  Name: Foo
+ *  Last name: Bar
+ *
+ */
+export let extractedTableValues = (errorTable: HTMLTableElement) => {
+  const headers = errorTable.getElementsByTagName('th');
+  const values = new Map();
+  for (const header of Array.from(headers)) {
+    const value = header.parentElement!.getElementsByTagName('td')[0].textContent;
+    values.set(header.textContent, typeof value === 'string' ? value.trim() : '');
+  }
+  return values;
+};
+
+/**
+ * Get string attribute and assert if the type is actually a string.
+ */
+export const getStringAttribute = curryN(2, (element: HTMLElement, name: string): string => {
```

Comment:
Can you please explain why you are using curryN as oposed to a 'regular' curry? I think in this situation the regular curry would work just fine. Problem is, I don't fully grasp in what situation one would use curryN

Reply:
Could you still provide the explanation?

Reply:
Nu gebruik ik curry